sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixClient.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl
+
+import akka.annotation.ApiMayChange
+import kalix.javasdk.{ DeferredCall, Metadata }
+import org.springframework.http.{ HttpHeaders, MediaType }
+import org.springframework.web.reactive.function.client.WebClient
+
+import java.util.concurrent.CompletionStage
+import scala.concurrent.{ Future, Promise }
+import scala.jdk.FutureConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+trait KalixClient {
+
+  @ApiMayChange
+  def post[P, R](uri: String, body: P, returnType: Class[R]): DeferredCall[P, R]
+
+  @ApiMayChange
+  def get[R](uri: String, resp: Class[R]): DeferredCall[_, R]
+
+}
+
+final class RestKalixClientImpl extends KalixClient {
+
+  private class RestDeferredCallImpl[P, R](asyncCall: () => CompletionStage[R]) extends DeferredCall[P, R] {
+    override def message(): P = ???
```

Comment:
Here is a place where we could add some clear error message to the users. 

This method will probably be called if we try to send this DeferredCall to the proxy. At some point, we will need to convert it to the a proto messages that is put on the wire.

We can throw an IllegalArgumentException saying that REST based DeferredCalls are not yet supported in forwards and timers.

Reply:
But we will need to test it out. We must make sure that the error will be propagated to user.

Reply:
Hmm, maybe better to not solve it with a runtime error, but let the compiler block it. 

We can for instance, keep `RestDeferredCallImpl`, but not let it extends `DeferredCall` for now. We keep the type though so users code will stay: `client.post("/url", Input, Output.class).execute()`. 

But users won't be able to pass it to `forward` and `timer`.

When we add the local transcoding we make it extend `DeferredCall` and everything will fall in its place.

Reply:
> Here is a place where we could add some clear error message to the users.

Yeah, I thought the same and had a IllegalArgumentException there but it was not really called anywhere and the user would still get the default unknown effect. In any case, agree that blocking at compile time is better.

So, to do that, the `KalixClient` API needs to return something different than a DeferredCall as well. Made it return a `RestDeferredCall` to avoid leaking the Impl.

(thanks for the suggestions Renato 🙇 )

Reply:
Correct, there is a chance that users will assign the `RestDeferredCall` to a variable, but that's the price for using experimental stuff. :-)

Reply:
Btw, let's keep it as internal API any.

Reply:
Got it, updated. Thanks again for all the tips 🙇